src/components/TableCreator/common/utils.ts

```diff
@@ -1,27 +1,34 @@
 import { ColumnDef } from "@tanstack/react-table";
-import { BaseColumnConfig } from "./entities";
+import { ColumnConfig } from "../../../config/entities";
 
 /**
  * Builds a base column definition.
+ * The following properties are set by default:
+ * - Grouping is not enabled `enableGrouping: false`.
+ * - Sorting is enabled - `enableSorting: true`.
  * @param baseColumnConfig - Base column configuration.
  * @returns column definition.
  */
-export function buildBaseColumnDef<T, TValue>(
-  baseColumnConfig: BaseColumnConfig<T, TValue>
+export function buildBaseColumnDef<T>(
+  baseColumnConfig: ColumnConfig<T>
 ): ColumnDef<T> {
   const {
     columnPinned,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- Destructure to avoid passing it to the columnDef.
+    columnVisible,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- Destructure to avoid passing it to the columnDef.
+    componentConfig,
     disableHiding,
-    disableSorting,
     header,
     id,
     meta,
     width,
+    ...columnDef
   } = baseColumnConfig;
   return {
+    ...columnDef,
     accessorKey: id,
     enableHiding: !disableHiding,
-    enableSorting: !disableSorting,
```

Comment:
`disableSorting` is removed from `ColumnConfig` in favour of TanStack API `enableSorting`.